### ISSUE TITLE
Multiple sales in same block can be conflated

### ISSUE THREAD

**Issue by fafrd** [crypt0biwan/curio-sales-bot#20]
These 3 sales occured in the same block, all from 0x86b to 0x426, initiated by 0x86b:

https://etherscan.io/tx/0x7340bc26618275c766ff0f25b77fe59b9cc7649c9ca2fd62858cd2bfcba18dc1
https://etherscan.io/tx/0xe52b18c1f942d49de1988a317b9e8bb53a319e1ccc1654485fb333a22dc5ec0c
https://etherscan.io/tx/0x3d8a45e5b151968ba38e96fcec069e77a5274b0c24a43dca8b0904f21647ce33

Each of these sell a different curio card, but the sales bot recognized them all as Card 4.

This is going to happen more frequently as more sales happen in the same block

**Comment by crypt0biwan** [comment]
Good catch, I'm sure this is caused by the way we handle "async" stuff.. Any idea how to solve this?
